integration-tests/utils/aws/cloudWatchGetLogs.ts

```diff
@@ -1,141 +1,116 @@
 import { cloudWatchLogsClient } from './cloudWatchLogsClient'
 import {
-  DescribeLogStreamsCommandInput,
-  DescribeLogStreamsCommandOutput,
   DescribeLogStreamsCommand,
   LogStream,
-  FilterLogEventsCommandInput,
   FilterLogEventsCommand,
-  FilterLogEventsCommandOutput,
   FilteredLogEvent
 } from '@aws-sdk/client-cloudwatch-logs'
 
-export const getLogStreamPrefix = () => {
-  const dateFormat = new Intl.DateTimeFormat('en-GB')
-  const dateParts = dateFormat.format(new Date()).split('/')
-  return `${
-    dateParts[2] + '/' + dateParts[1] + '/' + dateParts[0] + '/' + '[$LATEST]'
-  }`
-}
-
-const getLatestLogStreamName = async (): Promise<string> => {
-  const describeLogStreamsParams: DescribeLogStreamsCommandInput = {
-    logGroupName: INITIATE_DATA_REQUEST_LAMBDA_LOG_GROUP,
-    orderBy: 'LastEventTime',
-    descending: true,
-    limit: 1
-  }
-
-  const describeLogStreamsCommand = new DescribeLogStreamsCommand(
-    describeLogStreamsParams
+export const getCloudWatchLogEventsGroupByMessagePattern = async (
+  logGroupName: string,
+  eventMessagePatterns: string[],
+  maxAttempts = 30
+) => {
+  const event = await waitForEventWithPatterns(
+    logGroupName,
+    eventMessagePatterns,
+    maxAttempts
   )
 
-  await pause(500)
+  if (!event) return []
 
-  const describeLogStreamsResponse: DescribeLogStreamsCommandOutput =
-    await cloudWatchLogsClient.send(describeLogStreamsCommand)
+  const requestId = extractRequestIdFromEventMessage(event.message as string)
+  const eventLogStream = [{ logStreamName: event.logStreamName as string }]
+  const requestEndFilterPattern = [`END RequestId: ${requestId}`]
 
-  const latestLogStreams: LogStream[] =
-    describeLogStreamsResponse.logStreams ?? []
+  // Wait for final request in group
+  await waitForEventWithPatterns(
+    logGroupName,
+    requestEndFilterPattern,
+    maxAttempts
+  )
 
-  expect(latestLogStreams.length).toEqual(1)
-  return latestLogStreams[0].logStreamName as string
-}
+  const logEvents = await findMatchingLogEvents(logGroupName, eventLogStream, [
+    requestId
+  ])
+  console.log(`Events for Request Id ${requestId}:`, logEvents)
 
-interface EventLogStream {
-  logStreamName: string
-  eventMessage: string
+  return logEvents
 }
 
-export const waitForLogStreamContainingEvent = async (
-  eventFilterPattern: string,
-  ...eventMessagePatterns: string[]
-): Promise<EventLogStream> => {
-  if (eventMessagePatterns.length == 0) {
-    throw Error('No message patterns provided for event')
+const getLogStreams = async (logGroupName: string) => {
+  const input = {
+    logGroupName: logGroupName,
+    orderBy: 'LastEventTime',
+    descending: true,
+    limit: 5
   }
+  const command = new DescribeLogStreamsCommand(input)
+  const response = await cloudWatchLogsClient.send(command)
+  const logStreams = response?.logStreams ?? []
 
-  let latestLogStreamName = await getLatestLogStreamName()
```

Comment:
Is it okay to leave the old functions temporarily? I'll refactor to using them after I merge to my local. The athena events are a bit different to the initiate and process ones, and I'm not sure if the new ones would need more refactoring to get them working for those logs.

Reply:
Can we just pair to see why you would use this? I'm happy to help get things working. Plus if you really need it then just keep it in your branch.